src/tests/test_viewsGenerationSystem.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2021 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+"""
+Runs the dataView generation system on test data.
+Confirms the end result data is as expected.
+Creates a number of specific scenarios for running the transformation.
+"""
+
+from copy import deepcopy
+from enum import Enum
+import json
+import glob
+from logging import getLogger
+import os
+from pathlib import Path
+import shutil
+from src.tests.generateData import MockAddon
+from src.transform.datastructures import Addon, MajorMinorPatch, VersionCompatibility
+from typing import Iterable
+import subprocess
+import unittest
+
+log = getLogger()
+VIEWS_GLOB = "/**/**/*.json"
+
+
+class DATA_DIR(str, Enum):
+	_root = os.path.join(os.path.dirname(__file__), "test_data")
+	INPUT = os.path.join(_root, "input")
+	OUTPUT = os.path.join(_root, "output")
+	EXPECTED_RES = os.path.join(_root, "expected_results")
+	nvdaAPIVersionsPath = os.path.join(INPUT, "nvdaAPIVersions.json")
+
+
+class _TestDataGenerator:
+	_test_nvdaAPIVersions = [
+		VersionCompatibility(MajorMinorPatch(2020, 1), MajorMinorPatch(2020, 1)),
+		VersionCompatibility(MajorMinorPatch(2020, 2), MajorMinorPatch(2020, 1)),
+		VersionCompatibility(MajorMinorPatch(2021, 1), MajorMinorPatch(2021, 1)),
+	]
+
+	@staticmethod
+	def write_mock_addon_to_files(addon: Addon, exceptedVersions: Iterable[MajorMinorPatch]):
```

Comment:
I'm not sure what `exceptedVersions` means. Could you add these params to the doc string? Perhaps it's meant to be `expectedVersions` - the NVDA api versions that this addon should be the most recent release for`